tests/test_eventloop.py

```diff
@@ -0,0 +1,42 @@
+import math
+import sys
+
+import pytest
+
+from anyio import sleep_forever, sleep_until
+
+if sys.version_info < (3, 8):
+    from mock import AsyncMock
+else:
+    from unittest.mock import AsyncMock
+
+pytestmark = pytest.mark.anyio
+fake_current_time = 1620581544.0
+
+
+@pytest.fixture
+def fake_sleep(mocker):
+    mocker.patch('anyio._core._eventloop.current_time', return_value=fake_current_time)
+    return mocker.patch('anyio._core._eventloop.sleep', AsyncMock())
+
+
+async def test_sleep_until(fake_sleep):
+    sleep_time = 500.102352
+    deadline = fake_current_time + sleep_time
```

Comment:
it seems more suitable for this test to define `deadline`, and derive `sleep_time` from that (which doesn't need to be named)

```python3
deadline = fake_current_time + 500.102352
...
fake_sleep.assert_called_once_with(deadline - fake_current_time)
```